app/controllers/concerns/account_concern.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module AccountConcern
+  extend ActiveSupport::Concern
+
+  ACCOUNT_SESSION_HEADER_INTERNAL_NAME = "HTTP_GOVUK_ACCOUNT_SESSION"
+  ACCOUNT_SESSION_HEADER_NAME = "GOVUK-Account-Session"
+  ACCOUNT_END_SESSION_HEADER_NAME = "GOVUK-Account-End-Session"
+  ACCOUNT_SESSION_DEV_COOKIE_NAME = "govuk_account_session"
+
+  included do
+    before_action :fetch_account_session_header
+    before_action :set_account_session_header
+    before_action :set_account_variant
+
+    helper_method :logged_in?
+
+    attr_accessor :account_session_header
+  end
+
+  def logged_in?
+    account_session_header.present?
+  end
+
+  def fetch_account_session_header
+    @account_session_header =
+      if request.headers[ACCOUNT_SESSION_HEADER_INTERNAL_NAME]
+        request.headers[ACCOUNT_SESSION_HEADER_INTERNAL_NAME]
+      elsif request.headers.to_h[ACCOUNT_SESSION_HEADER_NAME]
+        request.headers.to_h[ACCOUNT_SESSION_HEADER_NAME]
+      elsif Rails.env.development?
+        cookies[ACCOUNT_SESSION_DEV_COOKIE_NAME]
+      end
+  end
+
+  def show_signed_in_header?
+    account_session_header.present?
+  end
```

Comment:
This is unused (the session controller doesn't return HTML), so it can be removed

Reply:
great thanks, removed